gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java

```diff
@@ -397,4 +410,48 @@ public Status status() {
   public int compareTo(MRCompactorJobRunner o) {
     return Double.compare(o.priority, this.priority);
   }
+
+  private List<Path> getCurLateEventPath(Path lateDataPath) throws FileNotFoundException, IOException {
+    List<Path> paths = Lists.newArrayList();
+    for (FileStatus fileStatus : FileListUtils.listFilesRecursively(fs, lateDataPath, new PathFilter() {
+      @Override
+      public boolean accept(Path path) {
+        for (String validExtention : getApplicableFileExtensions()) {
+          if (path.getName().endsWith(validExtention)) {
+            return true;
+          }
+        }
+        return false;
+      }
+    })) {
+      paths.add(fileStatus.getPath());
+    }
+    return paths;
+  }
+
+  /**
+   * Submit an event reporting new and cumulative late event counts.
+   */
+  private void submitLateEvent(List<Path> lateFilePaths, Path lateDataPath) {
+    if (this.jobProps.contains(MRCompactor.COMPACTION_HDFS_EVENT_COUNT_PROVIDER)) {
```

Comment:
Whether to submit late event reporting should not be determined by whether `jobProps` contains `MRCompactor.COMPACTION_HDFS_EVENT_COUNT_PROVIDER` because (1) it's not intuitive; (2) if I don't want to submit late event but want to do some other things with row counts, then my states may contain this property.